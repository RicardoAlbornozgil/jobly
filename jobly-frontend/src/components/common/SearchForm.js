import React, { useState } from "react";
import "../../styles/SearchForm.css";

/** Search widget.
 *
 * Appears on CompanyList and JobList so that these can be filtered
 * down.
 *
 * This component doesn't *do* the searching, but it renders the search
 * form and calls the `searchFor` function prop that runs in a parent to do the
 * searching.
 *
 * { CompanyList, JobList } -> SearchForm
 */

function SearchForm({ searchFor }) {
  console.debug("SearchForm", "searchFor=", typeof searchFor);

  const [searchTerm, setSearchTerm] = useState("");

  /** Tell parent to filter */
  function handleSubmit(evt) {
    evt.preventDefault();
    const trimmedTerm = searchTerm.trim();
    searchFor(trimmedTerm || undefined);
    setSearchTerm(trimmedTerm);
  }

  /** Update form fields */
  function handleChange(evt) {
    setSearchTerm(evt.target.value);
  }

  return (
    <div className="SearchForm">
      <form className="form-inline" onSubmit={handleSubmit}>
        <input
          className="form-control form-control-lg"
          name="searchTerm"
          placeholder="Enter search term..."
          value={searchTerm}
          onChange={handleChange}
          aria-label="Search"
        />
        <button type="submit" className="btn btn-lg btn-primary">
          Search
        </button>
      </form>
    </div>
  );
}

export default SearchForm;
