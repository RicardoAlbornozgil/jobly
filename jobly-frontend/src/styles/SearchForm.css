/* SearchForm.css */

.SearchForm {
  display: flex;
  width: 100%;
  justify-content: center; /* Center the form within the container */
  margin-bottom: 2rem; /* Add some space below the search form */
}

form.form-inline {
  display: flex;
  width: 100%;
  max-width: 600px; /* Limit the maximum width for better UX */
  flex-direction: row;
  justify-content: space-between; /* Distribute the space between input and button */
  align-items: center; /* Center items vertically */
}

.SearchForm input {
  flex-grow: 1; /* Allow the input to grow and fill available space */
  margin-right: 1rem; /* Add space between the input and the button */
  padding: 0.5rem 1rem; /* Add padding for better spacing */
  font-size: 1rem; /* Set a consistent font size */
  border-radius: 5px; /* Rounded corners for a modern look */
  border: 1px solid #ced4da; /* Light border color */
  transition: border-color 0.3s ease; /* Smooth transition for focus */
}

.SearchForm input:focus {
  box-shadow: 0 0 0 2px rgba(0, 123, 255, 0.25); /* Subtle blue shadow on focus */
  border-color: #80bdff; /* Change border color on focus */
}

.SearchForm button {
  padding: 0.5rem 1.5rem; /* Adjust button padding for a balanced look */
  font-size: 1rem; /* Consistent font size with input */
  border-radius: 5px; /* Match the button's border-radius with the input */
  transition: background-color 0.3s ease, transform 0.3s ease; /* Smooth transitions */
}

.SearchForm button:hover {
  background-color: #0056b3; /* Darker blue on hover */
  transform: scale(1.05); /* Slight scale up on hover */
}

.SearchForm button:active {
  background-color: #004085; /* Even darker blue when active */
  transform: scale(1); /* Reset to original size when active */
}
