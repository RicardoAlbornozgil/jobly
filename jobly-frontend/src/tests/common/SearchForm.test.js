import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/common/SearchForm.js";


it("matches snapshot", function () {
  const { asFragment } = render(<SearchForm />);
  expect(asFragment()).toMatchSnapshot();
});
