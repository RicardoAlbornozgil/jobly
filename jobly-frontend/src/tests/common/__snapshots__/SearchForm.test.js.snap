// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchForm"
  >
    <form
      class="form-inline"
    >
      <input
        aria-label="Search"
        class="form-control form-control-lg"
        name="searchTerm"
        placeholder="Enter search term..."
        value=""
      />
      <button
        class="btn btn-lg btn-primary"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
